### DISCUSSION TITLE
Infinite loop for keycloak?

### DISCUSSION THREAD

**Discussion by CJCShadowsan** [InseeFrLab/onyxia#390] [Q&A] [answered]
I raised it in the onyxia repo - But it seems i've got an infinite login loop whilst using keycloak.

I create the realm, I create the client, that all works as in I get redirected to the auth landing page, I log my details, it passes me back...

...But then it still says "login" at the top right, and if I click the catalog or my services, I get an infinite loop where it continually tries to loop over.

Any thoughts?

**Comment by garronej** [comment]
Hi @CJCShadowsan,  
Thank you for reaching out!  

How did you find Onyxia?  
What instruction did you follow to install it?  
https://docs.onyxia.sh/installing  

Best regards,

**Comment by CJCShadowsan** [comment]
Currently building out a stack, was looking at this because it's open 🙂

Was literally going through the step-by-step instructions to install via kubernetes on the old documentation, then have just gone through the docs listed.

Both times I get to the point where I auth, I check keycloak and it recognises me as a user session...

…Then I get redirected back to onyxia and it still thinks I have to login at the top right and if I click services catalog or my services, I get an infinite loop of retries.

Any ideas would be appreciated here - I’m clearly close!

**Reply by olevitt** [reply]
Hi,
No idea on top of my head but you are welcome to join the slack dedicated to onyxia : https://join.slack.com/t/3innovation/shared_invite/zt-19tht9hvr-bZGMdW8AV_wvd5kz3wRSMw
I suggest we setup a visio there to live debug. If you are ok with that it would be easier to pin down the issue you are facing

**Comment by garronej** [comment]
> Currently building out a stack, was looking at this because it's open 🙂
  
Nice :) You did well :)

> Both times I get to the point where I auth, I check keycloak and it recognises me as a user session...  

Is that so? 😣  
Would you copy paste your `onyxia-values.yaml` and `keycloak-values.yaml` here?  
I'll see if I can spot anything obvious.  
If not and in any case, you are very welcome to schedule a call with my college @olevitt.  

Best regards,

**Comment by CJCShadowsan** [comment, marked as answer]
For reference so I can mark this as an answer...:

Caused by:
![image](https://user-images.githubusercontent.com/16232240/195092060-8877f78a-7f87-4222-a25a-d76d96d12093.png)

Confidential setting not being correct. The moment I changed to public (thanks @olevitt for the call!), this stopped infinite looping.
![image](https://user-images.githubusercontent.com/16232240/195092004-35b54c67-1fd0-4062-b004-8ad7ccebce20.png)

Oddly enough keycloak recognised that it was an active session, onyxia just continually kept saying "this is a confidential token, not what i'm expecting" so kept redirecting to keycloak, which kept saying "You're logged in already, here's your token"... Ad infinitum 🤣 

In other news... Firefox's "slow everything down to a crawl via throttling for GPRS" is very useful!
![image](https://user-images.githubusercontent.com/16232240/195093222-b9668222-eb38-484d-aaf8-17cb23875b4c.png)

(and in future i'll be implementing opentelemetry with on UbiQ so i'll be able to capture this stuff in-flight with tracing)
